lib/fs/filesystem.go

```diff
@@ -77,8 +65,7 @@ type Filesystem interface {
 	SetXattr(path string, xattrs []protocol.Xattr, xattrFilter XattrFilter) error
 
 	// Used for unwrapping things
-	underlying() (Filesystem, bool)
-	wrapperType() filesystemWrapperType
```

Comment:
Gut feeling says we should go in the other direction and remove these from the public interface instead...

Reply:
@calmh this PR removes `wrapperType` from the public interface, are you proposing to also remove `Underlying`?

Reply:
I can imagine external implementations using a `basicFS` underneath, in which case I guess we'd want to expose `Underlying`.

Reply:
At that point, is it "external" or should it just be a PR? 🤔

Reply:
The question is whether an external Filesystem implementation could ever return anything from `Underlying` that is of interest to Syncthing. 

When implementing a completely standalone Filesystem (like in my cae an 'iOS Photos FS': no.

I can however imagine the following scenario: an external app wants to implement a sort of 'audit' filesystem that logs each and every FS call (like logFS). It would register a custom FS type that, when constructed, would instance a `basicFS` (using `NewFilesystem`, so not using `basicFS` directory. `NewFilesystem` could add wrappers to this instance). The custom FS should return that (possibly wrapped) fs from `Underlying` so it can be properly unwrapped later.

Reply:
I don't think that's a relevant use case and I don't want to lock in an API for it. I should say that every PR in this direction makes me more inclined to just move everything under `internal` in one swoop and say that any external API requires a significant discussion up front about what we want to support...

Reply:
I understand the reluctance to expose any further interfaces. Just to be clear, *I* still see these interfaces as unstable and know I need to do the work when anything at all changes (as evidenced by me tracking your work on `v2` and contributing some small changes to the external part of it). However I might not be the only user of this of course and I can understand wanting to avoid future complaints.

So then I will be abandoning this PR for now. A bit demotivating as I also want to move forward with the features that this was built for, but that's what it is.

I'd be happy to discuss and assist in developing an actual *stable* Go interface for embedding Syncthing if/when you are. The current `Internals` interface could be a starting point (but needs cleanup).

Reply:
So what I propose is that you move out the unexported methods to a different interface that we check for when needed internally, and that your external think wouldn't need to care about or implement. I didn't say you can't do your thing.

Reply:
Right, so that's an easy fix, back up to you then.

(Even still, the second part of your comment points to a valid and more fundamental discussion we should have at some point.)